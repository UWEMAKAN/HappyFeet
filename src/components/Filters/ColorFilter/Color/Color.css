.ColorItem {
  display: flex;
  align-items: center;
}

.ColorItem input {
  width: 1.6rem;
  height: 1.6rem;
}

.Color {
  height: 1.6rem;
  width: 1.6rem;
  border-radius: 2rem;
  margin: 0 1.5rem;
  border: .1px solid rgba(0,0,0,.4);
}

.ColorItem:not(:last-child) {
  margin-bottom: 1.5rem;
}
