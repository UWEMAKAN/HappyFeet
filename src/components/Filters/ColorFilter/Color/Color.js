import React from 'react';
import propTypes from 'prop-types';
import './Color.css';

const Color = ({ name, color, checkedHandler }) => (
  <div className="ColorItem">
    <input onChange={checkedHandler} name={name.toLowerCase()} type="checkbox" />
    <div
      style={{ backgroundColor: color }}
      className="Color"
    >
      {' '}
    </div>
    <div>{name}</div>
  </div>
);

Color.propTypes = {
  name: propTypes.string.isRequired,
  color: propTypes.string.isRequired,
  checkedHandler: propTypes.func.isRequired
};

export default Color;
