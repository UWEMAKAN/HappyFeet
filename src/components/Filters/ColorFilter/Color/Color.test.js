import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

const renderColor = (args) => {
  const defaultProps = {
    name: '',
    color: '',
    checkedHandler: jest.fn()
  };
  const props = { ...defaultProps, ...args };
  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<Color {...props} />);
};

describe('Color Component', () => {
  test('should render Color Component', () => {
    const name = 'blue';
    const color = 'blue';
    const checkedHandler = jest.fn();
    const wrapper = renderColor({ name, color, checkedHandler });
    wrapper.find('input').simulate('change');

    expect.assertions(2);
    expect(checkedHandler).toHaveBeenCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });
});
