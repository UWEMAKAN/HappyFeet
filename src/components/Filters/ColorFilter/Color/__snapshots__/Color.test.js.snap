// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Component should render Color Component 1`] = `ShallowWrapper {}`;
